### PR TITLE
Created UnityPerms Class

### PR DESCRIPTION
**Pull request by sheldor1510**
- #88

### CODE REVIEW

**Review by hakasapl — CHANGES_REQUESTED** [review]
Excellent work, I have 2 small change requests. I suspect we may revisit some methods here as we add more components of the group system and get a clearer idea.

File: resources/lib/UnityPerms.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            return false;
+        }
+
+        if (!$this->SQL->roleAvailableInGroup($uid, $group, $role)) {
+            return false;
+        }
+
+        $user_role = $this->SQL->getRole($uid, $group);
+
+        if ($this->SQL->hasPerm($user_role, 'unity.admin') || $this->SQL->hasPerm($user_role, 'unity.admin_no_grant')) {
```
[the remark this answers is not in the document]
Comment (sheldor1510) [reply resources/lib/UnityPerms.php:115]: Added

File: resources/lib/UnityPerms.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
```
[the remark this answers is not in the document]
Comment (sheldor1510) [reply resources/lib/UnityPerms.php:82]: Added check

**Review by hakasapl — APPROVED** [review]
Great work!

**Pull request merged**